### DISCUSSION TITLE
Conference on Android and IOS to HLS

### DISCUSSION THREAD

**Discussion by lexy10** [ant-media/Ant-Media-Server#6641] [Q&A]
We are trying to build something like tiktok or instagram video or voice rooms where users can join and leave. But we want to make use of the mobile SDKs and then stream it to HLS. We can't find a very good tutorial for that on the docs page.

Please has anyone done this before or have an idea on how to go about it?

**Comment by lastpeony** [comment]
Take a look at to the samples on SDK repos.
For android see Conference sample.

Each participant in conference is a broadcast in ant media server which means you can play participants with HLS using any HLS player on mobile. Its no different than playing a stream with HLS on mobile. Once stream is broadcasting state in ant media server it means you can play it from anywhere with any protocol. For webrtc play you need to utilize SDKs but for HLS play you can use other video player libraries such as:
https://github.com/androidx/media
See exoplayer

**Reply by lexy10** [reply]
I just checked it now, but it looks like the renderers are predefined, Can we have like a a dynamic layout depending on the number of publishers and then, when playing with hls, do I have to play each stream separately? This is also because of recording the stream... I'm really confused on the solution.

**Reply by lastpeony** [reply]
You only need to play the room broadcast.(mainTrack) for webrtc.
In hls play yes, you need to play them seperately.
